prov/mrail/src/mrail_cq.c

```diff
@@ -369,15 +369,19 @@ void mrail_poll_cq(struct util_cq *cq)
 	struct mrail_tx_buf *tx_buf;
 	struct fi_cq_tagged_entry comp;
 	fi_addr_t src_addr;
-	size_t i;
+	size_t i, idx;
 	int ret;
+	static int last_succ_rail = 0;
```

Comment:
Does this work when dealing with multiple peers, particularly if they have different default rails set?

Reply:
No. That can be improved.